src/client/react.js

```diff
@@ -56,7 +56,7 @@ export function useSession(options = {}) {
 
   React.useEffect(() => {
     if (requiredAndNotLoading) {
-      const url = `/api/auth/signin?${new URLSearchParams({
+      const url = `${_apiBaseUrl()}/signin?${new URLSearchParams({
```

Comment:
I think we could use `parseUrl()` rather than having duplicate utils?

Reply:
I am not sure on this one. 

As far as I understand, `_apiBaseUrl()` checks the `__NEXTAUTH` object, which in turn makes use of the `parseUrl()` function. So technically it could be refactored.

I would leave it on the core team to make decisions on this and refactor accordingly.

Reply:
Good point, I didn't realise `_apiBaseUrl()` calls `parseUrl()` internally, as you mention worth of a refactor in a different PR 👍🏽

Reply:
@mahieyin-rahmun could you move this change to a different PR please? 🙏🏽